src/chat_analyzer.py

```diff
@@ -158,15 +158,59 @@ def import_data(path_to_chatfile: str) -> List[Dict[str, Any]]:
     if isWa:
         print('Whatsapp chat recognized')
         f.seek(0)
+        #To check the dateformats
+        formatdmy = 1
+        formatmdy = 1
+        formatymd = 1
+        #To check that the date formats in the chat are in which format
+        #Can also check the time format here itself...
         for line in f:
             match = re.search(WMsg, line)
             if match:
+                try:
+                    datetime_object = datetime.strptime(match.groupdict()['date'], '%m/%d/%y')
+                except:
+                    formatmdy = 0
+                try:
+                    datetime_object = datetime.strptime(match.groupdict()['date'], '%d/%m/%y')
+                except:
+                    formatdmy = 0
+                try:
+                    datetime_object = datetime.strptime(match.groupdict()['date'], '%y/%m/%d')
+                except:
+                    formatymd = 0    
+        f.seek(0)
+        for line in f:
+            match = re.search(WMsg, line)
+            if match and formatmdy:
                 msgs.append({
                     'username': match.groupdict()['username'],
                     'date': datetime.strptime(match.groupdict()['date'], '%m/%d/%y').date(),
-                    'month': match.groupdict()['month'],
-                    'day': match.groupdict()['day'],
-                    'year': match.groupdict()['year'],
+                    'month': match.groupdict()['feild1'],
+                    'day': match.groupdict()['feild2'],
+                    'year': match.groupdict()['feild3'],
+                    'time': datetime.strptime(match.groupdict()['time'], '%H:%M').time(),
+                    'hour': match.groupdict()['hour'],
+                    'minute': match.groupdict()['minute'],
+                })
```

Comment:
Then, here, instead of having the if-else chain, we can just have the one if condition checking if there is a match or not, and if there is just use the parameters that were stored earlier.